tests/model_inheritance/test_abstract_inheritance.py

```diff
@@ -60,7 +61,7 @@ class Derived(DescendantOne, DescendantTwo):
 
         self.assertEqual(DescendantOne._meta.get_field('name').max_length, 30)
         self.assertEqual(DescendantTwo._meta.get_field('name').max_length, 50)
-        self.assertEqual(Derived._meta.get_field('name').max_length, 50)
+        self.assertEqual(Derived._meta.get_field('name').max_length, 30)
```

Comment:
This change isn't right. `Derived.mro()` is 

```
[Derived, 
 DescendantOne, 
 DescendantTwo, 
 AbstractBaseOne, 
 AbstractBaseTwo, 
 django.db.models.base.Model, 
 object,
]
```

The existing test is correct that `Derived` should inherit the `name` field from `DescendantTwo`. i.e. `max_length` should be 50.

Reply:
OK, so re-reading the description: 

> An update to an existing test, covering the new overriding order of DeferredAttributes. Arguably, what was previously being tested for didn't make sense.

So, running this change we have: 

```
DescendantOne.__dict__['name']
<django.db.models.query_utils.DeferredAttribute object at 0x102869278>
```

This explains the change in the test, **but** it's not what we expect from Python inheritance. So... we're trading one issue for another. 🤔

Reply:
OOOOH. Gross, you're right.
I would've assumed order is:
```
[Derived, 
 DescendantOne,
 AbstractBaseOne,
 AbstractBaseTwo,
 DescendantTwo,
 django.db.models.base.Model, 
 object,
]
```
This would be correct if `DescendantTwo` wasn't also inheriting from `AbstractBaseOne` and `AbstractBaseTwo`.

Thanks for looking at this again! Let me know what you think after you've thought about it some more, but I think you're right that the change in this line should be reverted.. I'll need to check why it's messed up the order then.. hrm.

Reply:
Some time later... 🙂

About to mail the list re whether this change is acceptable.